File: tests/transpiler/x/pas/inner_join.pas
{$mode objfpc}
program Main;
type Anon30 = record
  id: integer;
  name: string;
end;
type Anon31 = record
  id: integer;
  customerId: integer;
  total: integer;
end;
type Anon32 = record
  orderId: integer;
  customerName: string;
  total: integer;
end;
var
  customers: array of Anon30;
  orders: array of Anon31;
  result: array of Anon32;
  o: Anon31;
  c: Anon30;
  entry: integer;
begin
  customers := [(id: 1; name: 'Alice'), (id: 2; name: 'Bob'), (id: 3; name: 'Charlie')];
  orders := [(id: 100; customerId: 1; total: 250), (id: 101; customerId: 2; total: 125), (id: 102; customerId: 1; total: 300), (id: 103; customerId: 4; total: 80)];
  result := [];
  for o in orders do begin
  for c in customers do begin
  if o.customerId = c.id then begin
  result := concat(result, [(orderId: o.id; customerName: c.name; total: o.total)]);
end;
end;
end;
  writeln('--- Orders with customer info ---');
  for entry in result do begin
  writeln('Order', entry.orderId, 'by', entry.customerName, '- $', entry.total);
end;
end.

File: archived/tests/transpiler/x/pl/inner_join.pl
:- style_check(-singleton).
:- initialization(main).

main :-
    Customers = [{id: 1, name: "Alice"}, {id: 2, name: "Bob"}, {id: 3, name: "Charlie"}],
    Orders = [{id: 100, customerId: 1, total: 250}, {id: 101, customerId: 2, total: 125}, {id: 102, customerId: 1, total: 300}, {id: 103, customerId: 4, total: 80}],
    Result = [{orderId: 100, customerName: "Alice", total: 250}, {orderId: 101, customerName: "Bob", total: 125}, {orderId: 102, customerName: "Alice", total: 300}],
    writeln("--- Orders with customer info ---"),
    Entry = {orderId: 100, customerName: "Alice", total: 250},
    writeln("Order 100 by Alice - $ 250"),
    Entry1 = {orderId: 101, customerName: "Bob", total: 125},
    writeln("Order 101 by Bob - $ 125"),
    Entry2 = {orderId: 102, customerName: "Alice", total: 300},
    writeln("Order 102 by Alice - $ 300").
